pkg/scheduler/api/latest/latest.go

```diff
@@ -44,10 +45,11 @@ var Codec runtime.Codec
 
 func init() {
 	jsonSerializer := json.NewSerializer(json.DefaultMetaFactory, schedulerapi.Scheme, schedulerapi.Scheme, true)
```

Comment:
I know I'm late to this PR, but according to the godoc, `json.NewSerializer` returns an object that can parse JSON or YAML bytes. Are those docs wrong?

Reply:
Probably, it turns out to be working for both JSON **and** YAML.

Reply:
I guess I'm just surprised that it wasn't working before this PR, because the docs imply that it should have worked. Clearly @danielqsj would not have sent this unless it was broken.

Reply:
I checked it again.
After remove `yaml.NewDecodingSerializer`, unit test failed:
```
=== RUN   TestInitPolicyFromFile
--- FAIL: TestInitPolicyFromFile (0.00s)
    scheduler_test.go:1011: Failed writing a policy config file: invalid policy: couldn't get version/kind; json parse error: invalid character 'a' looking for beginning of value
FAIL
```

And the doc of `json.NewSerializer` is 
```
// NewSerializer creates a JSON serializer that handles encoding versioned objects into the proper JSON form. If typer
// is not nil, the object has the group, version, and kind fields set.
```

It seems the return object can only parse JSON, right?

So I turned to `yaml.NewDecodingSerializer` whose purpose is to support YAML decoding.
```
// NewDecodingSerializer adds YAML decoding support to a serializer that supports JSON.
func NewDecodingSerializer(jsonSerializer runtime.Serializer) runtime.Serializer {
	return &yamlSerializer{jsonSerializer}
}
```